src/SongManager.cpp

```diff
@@ -102,18 +110,15 @@ void SongManager::InitAll( LoadingWindow *ld )
 	{
 		m_GroupsToNeverCache.insert(*group);
 	}
-	InitSongsFromDisk( ld );
-	InitCoursesFromDisk( ld );
+	InitSongsFromDisk( ld, onlyAdded );
+	InitCoursesFromDisk( ld, onlyAdded );
+	if (onlyAdded)
+		DeleteAutogenCourses();

```

Comment:
I couldn't find any adverse effects, because there is no sort menu in course mode.

(braces added. I usually avoid them for single-line bodys because SMs style with the opening brace on it's own line takes so much space :rofl: )

Reply:
The adverse effect I was thinking of was something like:
- Someone plays an autogenned course
- Reloads courses (but I guess not possible in SL)
- Goes to ScreenEvalSummary but the original autogenned course was deleted so maybe crashes?

But yeah I do think it's something that likely won't come up so likely okay.

-------------

I think braces are good because it prevents oversights like:
```cpp
if (condition)
    function1();
    function2();
```
but yeaaaaah I do agree that SMs brace style makes it a bit annoying in that regard so /shrug